### PR TITLE
Support using the default application credentials for google client

### PR DESCRIPTION
**Pull request by zymap**
---

### Motivation

In the kubernetes environment, we more like use workload identity to all a pod to access the google service. Using the default application credentials will scan all possible credentails to access the google service.

### CODE REVIEW

File: catalog/files/impl/src/main/java/org/projectnessie/catalog/files/gcs/GcsClients.java [hunk 1]
```diff
@@ -148,7 +149,12 @@ static Credentials buildCredentials(
                 oauth2token.expiresAt().map(i -> new Date(i.toEpochMilli())).orElse(null));
         return OAuth2Credentials.create(accessToken);
       default:
-        throw new IllegalArgumentException("Unsupported auth type " + authType);
+        try {
```
Comment (snazy) [inline catalog/files/impl/src/main/java/org/projectnessie/catalog/files/gcs/GcsClients.java:152]: I'd rather leave the `default` case as it is but add a new enum value (e.g. `WORKLOAD_IDENTITY`) to `GcsBucketOptions.GcsAuthType`.

**Review by snazy — COMMENTED** [review]
@zymap Thank you for your contribution!

I've left a comment on the PR.

**Comment by adutra** [comment]
Related to #8987.

File: catalog/files/impl/src/main/java/org/projectnessie/catalog/files/gcs/GcsClients.java (hunk 1 above)
Comment (zymap) [reply catalog/files/impl/src/main/java/org/projectnessie/catalog/files/gcs/GcsClients.java:152]: Sure

Comment (zymap) [reply catalog/files/impl/src/main/java/org/projectnessie/catalog/files/gcs/GcsClients.java:152]: @snazy PTAL

**Review by snazy — APPROVED** [review]
LGTM

**Pull request merged**